source/slang/slang-lower-to-ir.cpp

```diff
@@ -5275,34 +5291,37 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         // a witness table for the interface type's conformance
         // to its own interface.
         //
-        List<IRStructKey*> requirementKeys;
+        NestedContext nestedContext(this);
+        auto subBuilder = nestedContext.getBuilder();
+        auto subContext = nestedContext.getContext();
+        List<IRInterfaceRequirementEntry*> requirementEntries;
+
         for (auto requirementDecl : decl->members)
         {
-            requirementKeys.add(getInterfaceRequirementKey(requirementDecl));
-
+            auto key = getInterfaceRequirementKey(requirementDecl);
+            auto entry = subBuilder->createInterfaceRequirementEntry(key, nullptr);
```

Comment:
This seems to be part of the problem, insofar as we are not creating the lowered type of the requirement at this point in the code, despite this being an obvious/appropriate time to do so. I'm confused about when this stuff gets wired up if it doesn't happen here.

It seems like all this code needs/wants to do is to lower the type of `requirementDecl` in an appropriately-construct `IRGenContext` - one that will map the AST `ThisType` -> `IRThisType`, and references to any associated types to `IRAssociatedType`. It should be possible to construct an appropriate `IRGenEnv`/`IRGenContext` before entering this loop.